src/test/java/org/rulez/demokracia/pdengine/VoteSetParametersTest.java

```diff
@@ -0,0 +1,179 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultVoteRegistry;
+
+public class VoteSetParametersTest extends CreatedDefaultVoteRegistry {
+
+	private Vote vote;
+	boolean canAddin, canEndorse, canVote, canView;
+	int minEndorsements;
+	
+	String originVoteId, originAdminKey;
+	List<String> originNeededAssurances, originCountedAssurances;
+	Boolean originIsPrivate;
+	long originCreationTime;
+	
+	@Before
+	public void setUp() throws ReportedException {
+		super.setUp();
+		vote = voteManager.getVote(adminInfo.voteId);
+		
+		originVoteId = vote.id;
+		originAdminKey = vote.adminKey;
+		originNeededAssurances = new ArrayList<String>(vote.neededAssurances);
+		originCountedAssurances = new ArrayList<String>(vote.countedAssurances);
+		originIsPrivate = vote.isPrivate;
+		originCreationTime = vote.creationTime;
+		
+		minEndorsements = 3;
+		canAddin = true;
+		canEndorse = true;
+		canVote = true;
+		canView = true;
+		voteManager.setVoteParameters(adminInfo.voteId, 
+				adminInfo.adminKey,
+				minEndorsements,
+				canAddin,
+				canEndorse,
+				canVote,
+				canView);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_voteId_is_rejected() {
+		String voteName = "modifiedVoteName";
+		assertThrows(
+				() -> voteManager.setVoteParameters(voteName,
+						adminInfo.adminKey,
+						minEndorsements,
+						canAddin,
+						canEndorse,
+						canVote,
+						canView)
+			).assertMessageIs(String.format("illegal voteId: %s", voteName));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_adminKey_is_rejected() {
+		String invalidAdminKey = RandomUtils.createRandomKey();
+		assertThrows(
+				() -> voteManager.setVoteParameters(adminInfo.voteId,
+						invalidAdminKey,
+						minEndorsements,
+						canAddin,
+						canEndorse,
+						canVote,
+						canView)
+			).assertMessageIs(String.format("Illegal adminKey: %s", invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("set vote parameters")
+	@tested_behaviour("vote invariants")
+	@Test
+	public void setVoteParameters_sets_the_minEndorsement_parameter_of_the_vote() {
```

Comment:
1. This is not the "vote invariants" behaviour. The most close behaviour is "update vote title" https://adadocs.demokracia.rulez.org/PDEngine/edemo/master/index.html#04c661ac-e467-430b-a514-9b8e6c84de5f
That behaviour directly contradicts to the tested behaviour.
However this time the model is wrong. I am modeling a "update vote parameters" behaviour shortly. Please use that.